core/src/main/java/org/lflang/util/ImportUtil.java

```diff
@@ -0,0 +1,99 @@
+package org.lflang.util;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.emf.ecore.resource.Resource;
+
+/**
+ * Utility class for handling package-related URIs in the context of LF (Lingua Franca) libraries.
+ * This class provides methods to build URIs for accessing library files based on their location in
+ * a project structure, specifically targeting the "target/lfc_include" directory for library
+ * inclusion.
+ */
+public class ImportUtil {
+
+  /**
+   * Builds a package URI based on the provided URI string and resource. It traverses upwards from
+   * the current resource URI until it finds the "src/" directory, then constructs the final URI
+   * pointing to the library file within the "target/lfc_include" directory.
+   *
+   * @param uriStr A string representing the URI of the file. It must contain both the library name
+   *     and file name, separated by a '/'.
+   * @param resource The resource from which the URI resolution should start.
+   * @return The constructed package URI as a string.
+   * @throws IllegalArgumentException if the URI string does not contain both library and file
+   *     names.
+   */
+  public static String buildPackageURI(String uriStr, Resource resource) {
+    URI currentURI = URI.createURI(".");
+    URI rootURI = currentURI.resolve(resource.getURI());
+    StringBuilder pathBuilder = new StringBuilder();
```

Comment:
Don't use a string builder to construct paths.

https://docs.oracle.com/javase/tutorial/essential/io/pathOps.html